src/vm/eventpipe.cpp

```diff
@@ -517,13 +529,19 @@ void QCALLTYPE EventPipeInternal::DeleteProvider(
 
 void QCALLTYPE EventPipeInternal::WriteEvent(
     INT_PTR eventHandle,
+    unsigned int eventID,
     void *pData,
     unsigned int length)
 {
     QCALL_CONTRACT;
     BEGIN_QCALL;
 
-    // TODO
+    _ASSERTE(eventHandle != NULL);
+    EventPipeEvent *pEvent = reinterpret_cast<EventPipeEvent *>(eventHandle);
+    BYTE *buffer = new BYTE[length];
```

Comment:
You can just pass pData directly to WriteEvent here.  It has the same lifetime as buffer here, so it's just an extra allocation.